integration-tests/hazelcast/src/main/java/org/apache/camel/quarkus/component/hazelcast/it/HazelcastRoutes.java

```diff
@@ -0,0 +1,189 @@
+    public static final String MOCK_MULTIMAP_DELETED = "mock:multimap-removed";
+    public static final String MOCK_REPLICATED_ADDED = "mock:replicatedmap-added";
+    public static final String MOCK_REPLICATED_DELETED = "mock:replicatedmap-removed";
+    public static final String MOCK_TOPIC_RECEIVED = "mock:topic-received";
+
+    @Inject
+    HazelcastInstance hazelcastInstance;
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-instance")
+    HazelcastDefaultComponent hazelcastInstance() {
+        final HazelcastInstanceComponent hazelcastComponent = new HazelcastInstanceComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-atomicvalue")
+    HazelcastDefaultComponent hazelcastAtomicnumber() {
+        final HazelcastAtomicnumberComponent hazelcastComponent = new HazelcastAtomicnumberComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-list")
+    HazelcastDefaultComponent hazelcastList() {
+        final HazelcastListComponent hazelcastComponent = new HazelcastListComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-map")
+    HazelcastDefaultComponent hazelcastMap() {
+        final HazelcastMapComponent hazelcastComponent = new HazelcastMapComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-multimap")
+    HazelcastDefaultComponent hazelcastMultimap() {
+        final HazelcastMultimapComponent hazelcastComponent = new HazelcastMultimapComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-replicatedmap")
+    HazelcastDefaultComponent hazelcastReplicatedmap() {
+        final HazelcastReplicatedmapComponent hazelcastComponent = new HazelcastReplicatedmapComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-set")
+    HazelcastDefaultComponent hazelcastSet() {
+        final HazelcastSetComponent hazelcastComponent = new HazelcastSetComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-topic")
+    HazelcastDefaultComponent hazelcastTopic() {
+        final HazelcastTopicComponent hazelcastComponent = new HazelcastTopicComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    private HazelcastDefaultComponent getHazelcastComponent(HazelcastDefaultComponent hazelcastComponent) {
```

Comment:
`configureHazelcastComponent` would perhaps better describe what this method does.